### PR TITLE
Add a non-polymorphic variant to RefCounted.

### PR DESCRIPTION
**Pull request by soheilhy**
Using RefCounted<Child, NonPolymorphic> users can now build smart,
ref-counted pointers without paying the costs of a vtable when it's
possible.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%   +1024 [None]                                                                               +15.7Ki  +0.2%
      +0.1%   +1024 [Unmapped]                                                                           +15.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%   +1024 TOTAL                                                                                +15.7Ki  +0.1%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +24  +0.0%

  [ = ]       0 TOTAL      +24  +0.0%



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%   +1024 [None]                                                                               +15.7Ki  +0.2%
      +0.1%   +1024 [Unmapped]                                                                           +15.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%   +1024 TOTAL                                                                                +15.7Ki  +0.1%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +24  +0.0%

  [ = ]       0 TOTAL      +24  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%   +1024 [None]                                                                               +15.7Ki  +0.2%
      +0.1%   +1024 [Unmapped]                                                                           +15.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%   +1024 TOTAL                                                                                +15.7Ki  +0.1%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +24  +0.0%

  [ = ]       0 TOTAL      +24  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,911      Total (>)      2,019,380

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,977,566       Core (>)      3,973,562

11,184,302      Total (>)     11,180,294


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,911      Total (>)      2,019,380

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,977,566       Core (>)      3,973,562

11,184,307      Total (>)     11,180,300


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: src/core/lib/gprpp/ref_counted_ptr.h [hunk 1]
```diff
@@ -102,8 +107,15 @@ class RefCountedPtr {
   }
 
   // If value is non-null, we take ownership of a ref to it.
+  void reset(T* value) {
+    if (value_ != nullptr) value_->Unref();
+    value_ = value;
+  }
+
```
[the remark this answers is not in the document]
Comment (soheilhy) [reply src/core/lib/gprpp/ref_counted_ptr.h]: Good point, done.

File: test/core/gprpp/ref_counted_test.cc [hunk 2]
```diff
@@ -29,20 +29,47 @@ namespace {
...
+class FooNonPolymorphic
+    : public RefCounted<FooNonPolymorphic, NonPolymorphicRefCount> {
+ public:
+  FooNonPolymorphic() {
+    static_assert(!std::has_virtual_destructor<FooNonPolymorphic>::value,
+                  "NonPolymorphicRefCount has a virtual dtor");
+  }
 };
 
 TEST(RefCounted, Basic) {
```
[the remark this answers is not in the document]
Comment (soheilhy) [reply test/core/gprpp/ref_counted_test.cc]: Sure, I moved them into separate TEST blocks.

File: src/core/lib/gprpp/ref_counted_ptr.h [hunk 3]
```diff
@@ -74,6 +75,8 @@ class RefCountedPtr {
   }
   template <typename Y>
   RefCountedPtr(const RefCountedPtr<Y>& other) {
+    static_assert(std::has_virtual_destructor<T>::value,
```
[the remark this answers is not in the document]
Comment (soheilhy) [reply src/core/lib/gprpp/ref_counted_ptr.h]: Yes, everything in type_traits is header-only  and compile-time. They would not add any instruction to the binary.

File: src/core/lib/gprpp/ref_counted.h [hunk 4]
```diff
@@ -34,14 +34,48 @@
...
 // New objects should be created via New() and start with a refcount of 1.
 // When the refcount reaches 0, the object will be deleted via Delete().
 //
 // This will commonly be used by CRTP (curiously-recurring template pattern)
 // e.g., class MyClass : public RefCounted<MyClass>
-template <typename Child>
-class RefCounted {
+//
+// Use PolymorphicRefCount and NonPolymorphicRefCount to select between
+// different implemetnation RefCounted.
```
[the remark this answers is not in the document]
Comment (soheilhy) [reply src/core/lib/gprpp/ref_counted.h]: thank you for catching this.

**Comment by soheilhy** [comment]
Thank you for the review @markdroth !

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%    +992 [None]                                                                               +15.7Ki  +0.2%
      +0.1%    +992 [Unmapped]                                                                           +15.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%    +992 TOTAL                                                                                +15.7Ki  +0.1%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/lib/gprpp/ref_counted.h [hunk 5]
```diff
@@ -56,7 +56,7 @@ class NonPolymorphicRefCount {
 // e.g., class MyClass : public RefCounted<MyClass>
 //
 // Use PolymorphicRefCount and NonPolymorphicRefCount to select between
-// different implemetnation RefCounted.
+// different implemetnations of RefCounted.
```
Comment (markdroth) [inline src/core/lib/gprpp/ref_counted.h]: s/implemetnations/implementations/

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/lib/gprpp/ref_counted.h (hunk 5 above)
Comment (soheilhy) [reply src/core/lib/gprpp/ref_counted.h]: Thank you for catching this! Sorry I made a stupid typo, not once but twice :P

File: src/core/lib/gprpp/ref_counted.h [hunk 6]
```diff
@@ -34,14 +34,58 @@
 
 namespace grpc_core {
 
+// PolymorphicRefCount enforces polymorphic destruction of RefCounted.
+class PolymorphicRefCount {
+ public:
+  GRPC_ABSTRACT_BASE_CLASS
```
Comment (soheilhy) [inline src/core/lib/gprpp/ref_counted.h]: @markdroth 
There is a compilation failures on Kokoru which doesn't happen on my machine. I had to add the following macros to fix it after your review:
`GRPC_ABSTRACT_BASE_CLASS` and `GPRC_ALLOW_CLASS_TO_USE_NON_PUBLIC_DELETE`

PTAL. Thank you

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%    +992 [None]                                                                               +16.7Ki  +0.2%
      +0.1%    +992 [Unmapped]                                                                           +16.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%    +992 TOTAL                                                                                +16.7Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/lib/gprpp/ref_counted.h (hunk 6 above)
Comment (markdroth) [reply src/core/lib/gprpp/ref_counted.h]: LGTM

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%    +992 [None]                                                                               +16.7Ki  +0.2%
      +0.1%    +992 [Unmapped]                                                                           +16.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%    +992 TOTAL                                                                                +16.7Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/lib/gprpp/ref_counted.h (hunk 6 above)
Comment (soheilhy) [reply src/core/lib/gprpp/ref_counted.h]: Thank you!

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,775      Total (>)      2,019,244

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,976,386       Core (>)      3,972,382

11,183,127      Total (>)     11,179,119


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,775      Total (>)      2,019,244

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,976,386       Core (>)      3,972,382

11,183,124      Total (>)     11,179,117


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by soheilhy** [comment]
Squashed the change and all failures were pre-existing. Will wait for another kokoru run.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.1%    +992 [None]                                                                               +16.7Ki  +0.2%
      +0.1%    +992 [Unmapped]                                                                           +16.7Ki  +0.2%
      [NEW]     +32 vtable for grpc_core::RefCounted<tsi::SslSessionLRUCache, grpc_core::PolymorphicRefC     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::(anonymous namespace)::MessageSizeLimits     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<grpc_core::RefCountedPtr<     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::SliceHashTable<std::unique_ptr<char, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::XdsLbClientStats, grpc_core::Polymorphic     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::GrpcLbClientStats, grpc_core::Polymorphi     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::FakeResolverResponseGenerator, grpc_core     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::channelz::BaseNode, grpc_core::Polymorph     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ServerRetryThrottleData, grpc_     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCounted<grpc_core::internal::ClientChannelMethodParams, grp     +32  [NEW]
      [NEW]     +32 vtable for grpc_core::RefCountedWithTracing<grpc_core::ConnectedSubchannel, grpc_cor     +32  [NEW]

  +0.1%    +992 TOTAL                                                                                +16.7Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by soheilhy** [comment]
Interop failure is due to #16692

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,775      Total (>)      2,019,244

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,976,386       Core (>)      3,972,382

11,183,120      Total (>)     11,179,121


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by soheilhy** [comment]
Failures are due to #16692

**Pull request merged**